File: 704-binary-search/704-binary-search.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var search = function(nums, target) {
    // initialize vars to represent left and right index values
    let [l, r] = [0, nums.length - 1];
    while (l <= r) {
        const m = Math.floor((r + l) / 2);
        if (nums[m] < target) l = m + 1;
        else if (nums[m] > target) r = m - 1;
        else return m;
    }
    return -1;
};

File: 704-binary-search/704-binary-search.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        r, l = len(nums) - 1, 0
        while l <= r:
            m = (l + r) // 2
            if nums[m] == target:
                return m
            elif nums[m] < target:
                l = m + 1
            else:
                r = m - 1
        return -1